src/processing/BoardState.java

```diff
@@ -0,0 +1,211 @@
+package processing;
+
+import java.util.*;
+import java.io.Serializable;
+import processing.utility.*;
+import processing.boardobject.*;
+
+/**
+* This BoardState class stores two maps which should be serializable for persistence.
+* ClientBoardState and ServerBoardState will store the object of this class in their class.
+*
+* @author Himanshu Jain, Shruti Umat
+* @reviewer Rakesh Kumar, Satchit Desai
+*/
+
+public class BoardState implements Serializable {
+
+	/**
+	 *  To authenticate that the user deserializing this object
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * Map from object ID to the Board Object
+	 */
+	private Map <ObjectId, BoardObject> objIdToBoardObject;
+
+	/**
+	 * Map from position on the board to Priority Queue of objects on that position
+	 */
+	private Map <Position, PriorityQueue <PriorityQueueObject>> posToObjects;
+
+	/**
+	 * Constructor to initialize both the maps
+	 */
+	public BoardState() {
+		objIdToBoardObject = new HashMap <ObjectId, BoardObject> ();
+		posToObjects = new HashMap <Position, PriorityQueue <PriorityQueueObject>> ();
+	}
+
+	/**
+	 * Synchronized handles to get and set the 'selectedObject' data member in ClientBoardState class
+	 */
+	public synchronized void setSelectedObject(PriorityQueueObject pqObject) {
+		ClientBoardState.selectedObject = new PriorityQueueObject(pqObject);
+	}
+
+	public synchronized PriorityQueueObject getSelectedObject() {
+		return ClientBoardState.selectedObject;
+	}
+
+	/**
+	 * Looks up BoardObject for the ObjectId key in the argument
+	 */
+	public synchronized BoardObject getBoardObjectFromId(ObjectId objId) {
+		return objIdToBoardObject.get(objId);
+	}
+
+	/**
+	 * Gets ObjectIds at the top of Priority Queues at each position (x, y) in given ArrayList.
+	 * The ObjectId that is present at the maximum number of positions i.e. which occurs
+	 * the maximum number of times is returned
+	 * This is done as the selected pixel positions can be such that
+	 * some of the pixels belong to an object and some to others.
+	 */
+	public synchronized ObjectId getMostProbableObjectId(ArrayList <Position> positions) {
```

Comment:
Function is Great !
But @himanshu2047  can you document the inner workings better

Reply:
no need to comment the whole code, sir said, just the important descriptions.

Reply:
It would be better for readers.....